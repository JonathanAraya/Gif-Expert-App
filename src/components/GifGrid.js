import React, {Fragment} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs';
// import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([])
    const { loading } = useFetchGifs();

    // useEffect(() => {
    //     getGifs( category )
    //         .then( imgs => setImages( imgs ) )
    // }, [ category ]);

    return (
        <Fragment>
            <h3>{ category }</h3>
            {loading ? 'Cargando...' : 'Finalizado'}
            {/* <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem
                            key={ img.id } 
                            { ...img }
                        />
                    ))
                }
        </div> */}
        </Fragment>
    )
}
