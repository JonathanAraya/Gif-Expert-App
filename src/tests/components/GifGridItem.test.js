import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GridItem />', () => {

    const title = 'Saludo';
    const url = 'https://localhost.cl/asd'
    const wrapper = shallow( <GifGridItem title={title} url={url}/> )

    test('Debe mostrar el componente correctamente ', () => {
        
        
        expect(wrapper).toMatchSnapshot();

    });
    
    test('debe tener un parrafo con el titulo', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    })

    test('debe tener una imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        //console.log(img.props('src'))

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('debe tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);

    })
    

});
