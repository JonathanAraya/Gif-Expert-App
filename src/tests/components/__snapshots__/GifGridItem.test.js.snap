// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GridItem /> Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Saludo"
    src="https://localhost.cl/asd"
  />
  <p>
    Saludo
  </p>
</div>
`;
